src/org/labkey/test/components/ui/ontology/OntologyTreeSearch.java

```diff
@@ -55,12 +58,76 @@ public OntologyTreeSearch selectItemWithLabel(String searchExpression, String la
         return this;
     }
 
-    private OntologyTreeSearch setInput(String value)
+    public boolean showsNoSearchResultsFound()
     {
+        return Locator.tagWithClass("div", "col").withText("No search results found.")
+                .existsIn(elementCache().resultContainer());
+    }
+
+    public boolean isResultContainerExpanded()
+    {
+        try
+        {
+            return elementCache().resultContainer().isDisplayed();
+        } catch (NoSuchElementException nse)
+        {
+            return false;
+        }
+    }
+
+    public OntologyTreeSearch setInput(String value)
+    {
+        clearInput();
         elementCache().searchInput.set(value);
         return this;
     }
 
+    /**
+     * sets the input text to empty and waits for the search results container to collapse
+     * and for the placeholder to become visible
+     * @return
+     */
+    public OntologyTreeSearch clearInput()
+    {
+        var input = elementCache().searchInput;
+        String placeholder = input.getComponentElement().getAttribute("placeholder");
+        input.set("");
+        // wait for the results container to collapse, and for the placeholder for the input to be shown
+        getWrapper().waitFor(()-> !isResultContainerExpanded() &&
+                Locator.css("input:placeholder-shown").withAttribute("placeholder", placeholder)
+                        .existsIn(this),  2000);
+        return this;
+    }
+
+    public List<TreeSearchResult> waitForResults()
+    {
+        getWrapper().waitFor(()-> isResultContainerExpanded() && getSearchResults().size() > 0 || showsNoSearchResultsFound()
+                , WebDriverWrapper.WAIT_FOR_JAVASCRIPT);
+        return getSearchResults();
+    }
+
+    private List<TreeSearchResult> getSearchResults()
+    {
+        return new TreeSearchResult.TreeSearchResultFinder(getDriver()).findAll(elementCache().resultContainer());
+    }
+
+    /**
+     * when more search hits exist than are shown (max of 20 can be shown), an added search result footer element
+     * will appear with text advising the user to refine their search
+     * @return
+     */
+    Optional<WebElement> searchResultFooterElement()
+    {
+        return Locator.tagWithClass("div", "result-footer").findOptionalElement(elementCache().resultContainer());
+    }
+
```

Comment:
The search result footer appears only when there are more search 'hits' than can be shown.  (the control will show a maximum of 20, and then it will show this footer element to advise the user to refine their search).